charts/assets/templates/deployment.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "assets.fullname" . }}
+  labels:
+    {{- include "assets.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "assets.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+    {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      labels:
+        {{- include "assets.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: {{ include "assets.serviceAccountName" . }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.version | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: DRIVE_NAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "assets.fullname" . }}-config
+                  key: driveName
+            - name: DRIVE_SCOPE
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "assets.fullname" . }}-config
+                  key: scope
+            - name: ROOT_FOLDER_ID
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "assets.fullname" . }}-config
+                  key: rootFolderID
+            - name: DRIVE_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "assets.fullname" . }}-config
+                  key: token
+            - name: GITHUB_BOT_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "assets.fullname" . }}-config
+                  key: github
+          volumeMounts:
+          - name: assets-data-volume
+            mountPath: /assets
+      volumes:
+      - name: config-volume
+        secret:
+          secretName: {{ include "assets.fullname" . }}-config
+      - name: assets-data-volume
+{{ if eq .Values.environment "ci" }}
+        emptyDir: {}
+{{ else }}
+        persistentVolumeClaim:
```

Comment:
Let's please use a volumeClaimTemplate inside the statefulset manifest, we don't need to use the pvc elsewhere see https://github.com/w3f/kusama-faucet/blob/master/charts/kusama-faucet/templates/statefulset.yaml for an example.